packages/xdl/src/Webpack.ts

```diff
@@ -93,30 +105,51 @@ export async function startAsync(
     !!process.stdout.isTTY
   );
 
-  const server: WebpackDevServer = await new Promise(resolve => {
-    // Create a webpack compiler that is configured with custom messages.
-    const compiler = createWebpackCompiler({
+  let server: DevServer;
+  if (usingNextJs) {
+    server = await new Promise(resolve => {
```

Comment:
```suggestion
    server = await startNextJsAsync({
```
Async functions should usually be called with `await` so that the code waits for the async function to execute. This also ensures errors get handled instead of becoming unhandled promise rejections.

Please also remove the `onFinished` callback and instead make the `startNextJsAsync` function wait until starting the server has finished. Async functions and callbacks are both programming techniques for asynchronous code. In this codebase we strongly prefer async functions (promises) over callbacks. You can use the `await new Promise(..)` construct inside `startNextJsAsync` to convert from callbacks to promises.